src/components/Page/TodoList.tsx

```diff
@@ -0,0 +1,190 @@
+import classNames from 'classnames';
+import React, {
+  FormEvent,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { TodosError } from '../../types/ErrorEnum';
+import { Todo } from '../../types/Todo';
+import { TodoContext } from '../TodoContext';
+
+type Props = {
+  visibleTodos: Todo[];
+  removeTodo: (todoId: number) => Promise<void>;
+  isAdding: boolean;
+  handleStatus: (todoId: number, data: Partial<Todo>) => Promise<void>;
+  setIsAdding: React.Dispatch<React.SetStateAction<boolean>>;
+  newTodoField: React.RefObject<HTMLInputElement>;
+  setTodosError: (value: React.SetStateAction<TodosError>) => void;
+  upgradeTodos: (todoId: number, data: Partial<Todo>) => Promise<void>;
+  input: string;
+};
+
+export const TodoList: React.FC<Props> = ({
+  visibleTodos,
+  removeTodo,
+  isAdding,
+  handleStatus,
+  setIsAdding,
+  newTodoField,
+  setTodosError,
+  upgradeTodos,
+  input,
+}) => {
+  const [deletedId, setDeletedId] = useState<number | null>(null);
+  const [todos, setTodos] = useContext(TodoContext);
+  const handleDelete = (todoId: number) => {
+    setDeletedId(todoId);
+    removeTodo(todoId);
+  };
+
+  const [isDoubleClick, setIsDoubleClick] = useState<boolean>(false);
+
+  const [name, setName] = useState<string>('');
```

Comment:
There no need to write type for primitives.
```suggestion
  const [isDoubleClick, setIsDoubleClick] = useState(false);

  const [name, setName] = useState('');
```